### PR TITLE
Make Owls Tameable

### PR DESCRIPTION
**Pull request by ncorder**
**Description**: Allows Owls to be tamed

Also adds following, drops, and natural spawning.

### CODE REVIEW

**Comment by ncorder** [comment]
I made this pr for: https://github.com/BlockySurvival/mobs_animal_pack/pull/4

However, I thought the PR might be of importance to upstream issues as well.

File: mobs/owl.lua [hunk 1]
```diff
@@ -1,46 +1,85 @@
...
-	passive = true,
-	hp_min = 12,
-	hp_max = 22,
-	armor = 130,
-	collisionbox = {-0.3, -0.5, -0.3, 0.2, 0.2, 0.2},
-	visual = "mesh",
-	mesh = "owl.b3d",
-	textures = {
-		{"dmobs_owl.png"},
+mobs:register_mob("mobs_doomed:owl", {
```
Comment (taikedz) [inline mobs/owl.lua]: Please change namespace back to `dmobs:*`
Changing this namespace here may break compatibility in other mobs depending on this one.

File: mobs/owl.lua [hunk 2]
```diff
@@ -1,46 +1,85 @@
...
+                "mobs_animals:rat",
+                "mobs_animals:bee",
+                "mobs_animals:bunny",
+                "mobs_animals:chicken",
+                "mobs_animals:kitten",
+                "mobs_fish:tropica",
+                "mobs_fish:clownfish",
+                "mobs_bat:bat",
+                "mobs_birds:bird_sm",
+                "mobs_doomed:hedgehog",
```
Comment (taikedz) [inline mobs/owl.lua]: Please leave the hdgehog under the `dmobs:*` namespace as well - owl will not follow any hedgehog with this dependency as-stated!

File: mobs/owl.lua [hunk 3]
```diff
@@ -1,46 +1,85 @@
...
+            return
+        end
+		if mobs:protect(self, clicker) then return end
+        mobs:capture_mob(self, clicker, 0, 5, 50, false, nil)
+    end,
 })
 
+mobs:register_egg("mobs_doomed:owl", "Owl", "default_tree.png", 1)
+
+mobs:spawn_specific("mobs_doomed:owl",
```
Comment (taikedz) [inline mobs/owl.lua]: PLease retain original namespace 😉

**Review by taikedz — CHANGES_REQUESTED** [review]
Hello @ncorder - thank you for the contribution!

An excellent suggestion, and thank you for the tidy-up as well.

I presume you changed your copy of `dmobs` for compatibility on your own server to a new namespace `mobs_doomed:*` ; however in this repo we wish to retain backwards compatibility with existing dependencies, hence the comments to retain the original namespace.

Thanks!

File: mobs/owl.lua (hunk 1 above)
Comment (ncorder) [reply mobs/owl.lua]: I believe you may be looking at an outdated comit, I forced ammended it.

Comment (ncorder) [reply mobs/owl.lua]: ah, fiddle sticks. That is on me, I pushed to the wrong branch. the most up to date commit should work.

File: mobs/owl.lua [hunk 4]
```diff
@@ -1,46 +1,84 @@
...
+    hp_min = 12,
+    hp_max = 22,
+    armor = 130,
+    collisionbox = {-0.3, -0.5, -0.3, 0.2, 0.2, 0.2},
+    visual = "mesh",
+    mesh = "owl.b3d",
+    textures = {
+        {"dmobs_owl.png"},
+    },
+    mobs:register_mob("dmobs:owl", {
```
Comment (taikedz) [inline mobs/owl.lua:13]: 🛑  This line looks suspiciously out of place ...

File: mobs/owl.lua [hunk 5]
```diff
@@ -1,46 +1,84 @@
...
+    visual_size = {x=2, y=2},
+    makes_footstep_sound = false,
+    walk_velocity = 0,
+    run_velocity = 0,
+    jump = false,
+    water_damage = 2,
+    lava_damage = 2,
+    light_damage = 0,
+    view_range = 1,
+   	follow = {
```
Comment (taikedz) [inline mobs/owl.lua:24]: ⚠️ Extra indent by 1 space?

File: mobs/owl.lua [hunk 6]
```diff
@@ -1,46 +1,84 @@
...
+		"farming:seed_wheat",
+		"farming:seed_cotton",
+		"mobs_bugslive:bug",
+		"mobs_butterfly:butterfly",
+		"butterflies:butterfly_red",
+		"butterflies:butterfly_violet",
+		"butterflies:butterfly_white",
+		"fireflies:firefly",
+                "mobs_animals:rat",
```
Comment (taikedz) [inline mobs/owl.lua:33]: ⚠️ Extra indents hint at addded code blocks - visual complexity during reading. Not a breaker, but not ideal ...

**Review by taikedz — CHANGES_REQUESTED** [review]
You seem to have a 🛑 bug 

I added one comment about ⚠️ indentation which is reallyt just stylistic. There are other places in which indentation is a bit off - has your editor converted some indentation into tabs instead of spaces? Would be good to retain unified indentation around 4 spaces per indentation please 😄

**Comment by mckaygerhard** [comment]
please close this @taikedz  in flavor of https://github.com/minetest-mobs-mods/dmobs/pull/8 that was tested and follows the code checks

**Pull request closed without merging**